### PR TITLE
Rename getPartValue to getCommittedValue, etc.

### PR DESCRIPTION
**Pull request by justinfagnani**
`getPartValue()` → `getCommittedValue()`
`setPartValue()` → `setCommittedValue()`
`Part._$value`  → `Part._$committedValue`

### CODE REVIEW

File: packages/lit-html/src/directive-helpers.ts [hunk 1]
```diff
@@ -149,31 +149,33 @@ export const setPartValue = <T extends Part>(
 const RESET_VALUE = {};
 
 /**
- * Resets the stored state of a ChildPart used for dirty-checking and
- * efficiently updating the part to the given value, or when omitted, to a state
- * where any subsequent dirty-check is guaranteed to fail, causing the next
- * commit to take effect.
+ * Resets the committed value a ChildPart directly without triggering the
```
Comment (sorvell) [inline packages/lit-html/src/directive-helpers.ts:152]:
```suggestion
 * Resets the committed value of a ChildPart directly without triggering the
```

File: packages/lit-html/src/lit-html.ts [hunk 2]
```diff
@@ -759,16 +759,15 @@ class TemplateInstance {
...
-      }
-      if ((part as AttributePart).strings !== undefined) {
-        (part as AttributePart)._$setValue(values, part as AttributePart, i);
-        i += (part as AttributePart).strings!.length - 1;
-      } else {
-        (part as ChildPart)._$setValue(values[i++]);
+      if (part !== undefined) {
+        if ((part as AttributePart).strings !== undefined) {
+          (part as AttributePart)._$setValue(values, part as AttributePart, i);
+          i += (part as AttributePart).strings!.length - 2;
```
Comment (sorvell) [inline packages/lit-html/src/lit-html.ts:765]: Ideally comment the `-2` here

File: packages/lit-html/src/directive-helpers.ts [hunk 3]
```diff
@@ -149,31 +149,33 @@ export const setPartValue = <T extends Part>(
...
+ * commit stage of the part.
+ *
+ * This is useful in cases where a directive needs to update the part such
+ * that the next update detects a value change or not.
```
Comment (kevinpschaaf) [inline packages/lit-html/src/directive-helpers.ts:156]: Add back the note describing the effect of omitting `value`? 

Something like:
> When `value` is omitted, the next update will be guaranteed to be detected as a change.

**Review by sorvell — APPROVED** [review]
LGTM, just a couple nits.

**Review by kevinpschaaf — APPROVED** [review]
Minor API docs nit

File: packages/lit-html/src/directive-helpers.ts (hunk 1 above)
Comment (justinfagnani) [reply packages/lit-html/src/directive-helpers.ts:152]: done

File: packages/lit-html/src/lit-html.ts (hunk 2 above)
Comment (justinfagnani) [reply packages/lit-html/src/lit-html.ts:765]: done

File: packages/lit-html/src/directive-helpers.ts (hunk 3 above)
Comment (justinfagnani) [reply packages/lit-html/src/directive-helpers.ts:156]: done

**Pull request merged**